src/corehost/common/utils.cpp

```diff
@@ -87,3 +114,59 @@ void replace_char(pal::string_t* path, pal::char_t match, pal::char_t repl)
         (*path)[pos] = repl;
     }
 }
+
+const pal::char_t* get_arch()
+{
+#if _TARGET_AMD64_
+    return _X("x64");
+#elif _TARGET_X86_
+    return _X("x86");
+#else
+#error "Unknown target"
+#endif
+}
+
+pal::string_t get_own_rid()
+{
+#if defined(COREHOST_RID)
+    return COREHOST_RID;
+#else
+    return _X("win10-x64");
```

Comment:
We should not have the "#else" case and always have the host RID. how are we getting COREHOST_RID?